### ISSUE TITLE
Erroneous missing-function-docstring with Generics

### ISSUE THREAD

**Issue by efroemling** [PyCQA/pylint#3605]
This might be related to existing bugs such as https://github.com/PyCQA/pylint/issues/2568,
but I'm running into some false positive errors when using generics with class hierarchies.

### Steps to reproduce
Run pylint on a module containing the following code:
```python
"""Module docs."""
# pylint: disable=too-few-public-methods, invalid-name

from typing import Generic, TypeVar

T = TypeVar('T')


class Class1(Generic[T]):
    """Class Docs."""
    def method1(self):
        """Method Docs."""


class Class2(Class1[T]):
    """Class Docs."""


class Class3a(Class2[T]):
    """Class Docs."""
    def method1(self):
        pass


class Class3b(Class2):
    """Class Docs."""
    def method1(self):
        pass


class Class3c(Class1[T]):
    """Class Docs."""
    def method1(self):
        pass
```

### Current behavior
Although a doc-string is defined for method1 in Class1, which Class3a inherits from, Pylint reports
`test.py:23:4: C0116: Missing function or method docstring (missing-function-docstring)`

Class3b and Class3c show two changes that will make the error go away (removing the generic parameter on Class2 and inheriting directly from Class1, respectively)

### Expected behavior
No missing-docstring complaints.

### pylint --version output
pylint 2.5.2
astroid 2.4.1
Python 3.7.7 (default, Mar 10 2020, 15:43:33) 
[Clang 11.0.0 (clang-1100.0.33.17)]

**Comment by mathiasertl** [comment]
Just sharing what I found while debugging that same issue before I found this github report, in the hopes that it helps solving the issue ;-). 

It appears (to me), that this is related to using an intermediate class. For example, this works fine:

```
# pylint: disable=missing-module-docstring,missing-class-docstring,too-few-public-methods

from typing import Generic
from typing import TypeVar

Test = TypeVar('Test')

class Parent(Generic[Test]):
    def test(self, arg):
        """test"""

class Child(Parent[Test]):
    def test(self, arg):
        # no pylint error
        pass
```

HOWEVER,  adding an intermediate class in between triggers the error:

```
# pylint: disable=missing-module-docstring,missing-class-docstring,too-few-public-methods

from typing import Generic
from typing import TypeVar

Test = TypeVar('Test')

class Parent(Generic[Test]):
    def test(self, arg):
        """test"""


class Intermediate(Parent[Test]):
    pass


class Child(Intermediate[Test]):
    def test(self, arg):
        # triggers C0116: Missing function or method docstring (missing-function-docstring)
        pass
```

Doing the same without using generic classes also does not yield the error, of course. I tried various things related to type annotations (e.g. using `Intermediate[int]`, actually annotating functions, ...) and I always get the error.

**Comment by mathiasertl** [comment]
This also appears to trigger other false warnings, for example `too-few-public-methods`:

```
# pylint: disable=missing-module-docstring,missing-class-docstring

from typing import Generic
from typing import TypeVar

Test = TypeVar('Test')


class Parent(Generic[Test]):
    def func_a(self):
        """documented."""

    def func_b(self):
        """Also documented."""

    def test(self, arg):
        """test"""


class Intermediate(Parent[Test]):
    pass


class Child(Intermediate[int]):
   # triggers R0903: Too few public methods (1/2) (too-few-public-methods)

    def test(self, arg):
        # triggers C0116: Missing function or method docstring (missing-function-docstring)
        pass
```
